release-notes/6.0/api-diff/Asp.Net/6.0.0_Microsoft.AspNetCore.Http.Metadata.md

```diff
@@ -0,0 +1,33 @@
+# Microsoft.AspNetCore.Http.Metadata
+
+``` diff
++namespace Microsoft.AspNetCore.Http.Metadata {
++    public interface IAcceptsMetadata {
++        IReadOnlyList<string> ContentTypes { get; }
++        bool IsOptional { get; }
++        Type RequestType { get; }
++    }
++    public interface IFromBodyMetadata {
++        bool AllowEmpty { get; }
++    }
++    public interface IFromHeaderMetadata {
++        string Name { get; }
++    }
++    public interface IFromQueryMetadata {
++        string Name { get; }
++    }
++    public interface IFromRouteMetadata {
++        string Name { get; }
++    }
++    public interface IFromServiceMetadata
```

Comment:
Looks like whatever generated this doesn't support marker interfaces properly?

Reply:
@safern probably owns this tool now.

Reply:
Thanks, @martincostello, yeah the tool doesn't support a lot of new C# features as it is written on CCI and we haven't implemented a ton of those. However with the new API Compat object model we've been writing on top of Roslyn moving this tool on top of that object model, it might be straightforward to get api diff moved to roslyn and get these new features for free. 

We also don't support function pointers for example.